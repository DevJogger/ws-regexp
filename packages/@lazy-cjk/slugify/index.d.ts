/**
 * Created by user on 2020/5/30.
 */
import { IOptionsSlugify } from './lib/types';
export declare function transliterate(word: string, options?: IOptionsSlugify): string;
export declare function slugify(word: string, options?: IOptionsSlugify): string;
export default slugify;
