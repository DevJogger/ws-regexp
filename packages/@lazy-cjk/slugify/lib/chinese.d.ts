import { IOptionsSlugify } from './types';
export declare function _replaceChinese(text: string, options?: IOptionsSlugify): string;
export declare function slugifyChinese(word: string, options?: IOptionsSlugify): string;
