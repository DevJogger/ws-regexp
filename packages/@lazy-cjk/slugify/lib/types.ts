import { EnumPinyinMode, EnumPinyinType } from '@lazy-cjk/cns-11643/lib/types';

export interface IOptionsSlugify
{
	separatorRegexp?: RegExp,
	/**
	 * @default `-`
	 */
	separator?: string,

	noSepBetweenZhChar?: boolean,
	noStripOthers?: boolean,

	deburr?: boolean,

	upperCaseExtra?: boolean,
	lowerCaseExtra?: boolean,
	lowerCase?: boolean,
	upperFirst?: boolean,
	upperCase?: boolean,

	trimRegexp?: RegExp,

	allowEmptyResult?: boolean,

	emoji?: boolean,
	/**
	 * @default true
	 */
	transliterate?: boolean,
	/**
	 * @default true
	 */
	cjk?: boolean,

	maxLength?: number,

	chineseOptions?: {
		useTransliteration?: boolean,
		pinyinMode?: EnumPinyinMode,
		pinyinType?: EnumPinyinType,
	},

	/**
	 * @default true
	 * @see {import('@lazy-cjk/korean-romanize').IOptionsRomanize}
	 */
	ignoreUnSupported?: boolean,
}

