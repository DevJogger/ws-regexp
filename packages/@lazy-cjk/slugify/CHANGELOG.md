# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.0.70](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.69...@lazy-cjk/slugify@1.0.70) (2020-09-01)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.69](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.68...@lazy-cjk/slugify@1.0.69) (2020-08-31)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.68](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.67...@lazy-cjk/slugify@1.0.68) (2020-08-26)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.67](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.66...@lazy-cjk/slugify@1.0.67) (2020-08-17)


### ‚ôªÔ∏è„ÄÄChores

* update deps ([bb52b67](https://github.com/bluelovers/ws-regexp/commit/bb52b67c5504e766875a6f7447112c7884dd7754))





## [1.0.66](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.65...@lazy-cjk/slugify@1.0.66) (2020-08-13)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.65](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.64...@lazy-cjk/slugify@1.0.65) (2020-08-11)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.64](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.63...@lazy-cjk/slugify@1.0.64) (2020-08-11)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.63](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.62...@lazy-cjk/slugify@1.0.63) (2020-08-03)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.62](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.61...@lazy-cjk/slugify@1.0.62) (2020-08-02)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.61](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.60...@lazy-cjk/slugify@1.0.61) (2020-08-01)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.60](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.59...@lazy-cjk/slugify@1.0.60) (2020-07-27)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.59](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.58...@lazy-cjk/slugify@1.0.59) (2020-07-24)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.58](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.57...@lazy-cjk/slugify@1.0.58) (2020-07-23)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.57](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.56...@lazy-cjk/slugify@1.0.57) (2020-07-19)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.56](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.55...@lazy-cjk/slugify@1.0.56) (2020-07-19)


### üîñ„ÄÄMiscellaneous

* . ([c3c7f7f](https://github.com/bluelovers/ws-regexp/commit/c3c7f7fc30adc9cd3fc116cc5cf11a0cc0911e16))





## [1.0.55](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.54...@lazy-cjk/slugify@1.0.55) (2020-07-19)


### ‚ôªÔ∏è„ÄÄChores

* update deps ([aee28f2](https://github.com/bluelovers/ws-regexp/commit/aee28f2539c01b5d19f5ea4fa6909a1e30719945))





## [1.0.54](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.53...@lazy-cjk/slugify@1.0.54) (2020-07-08)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.53](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.52...@lazy-cjk/slugify@1.0.53) (2020-07-07)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.52](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.51...@lazy-cjk/slugify@1.0.52) (2020-07-06)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.51](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.50...@lazy-cjk/slugify@1.0.51) (2020-07-05)


### üîñ„ÄÄMiscellaneous

* . ([cdd46e9](https://github.com/bluelovers/ws-regexp/commit/cdd46e9c06c49e19a6912962aef6be1716056cc0))





## [1.0.50](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.49...@lazy-cjk/slugify@1.0.50) (2020-07-05)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.49](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.48...@lazy-cjk/slugify@1.0.49) (2020-07-04)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.48](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.47...@lazy-cjk/slugify@1.0.48) (2020-07-04)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.47](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.46...@lazy-cjk/slugify@1.0.47) (2020-07-03)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.46](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.45...@lazy-cjk/slugify@1.0.46) (2020-07-02)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.45](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.44...@lazy-cjk/slugify@1.0.45) (2020-06-30)


### üîñ„ÄÄMiscellaneous

* . ([0551f64](https://github.com/bluelovers/ws-regexp/commit/0551f64ad78a7c512f503f2c11ab5e0973af7a1f))





## [1.0.44](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.43...@lazy-cjk/slugify@1.0.44) (2020-06-30)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.43](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.42...@lazy-cjk/slugify@1.0.43) (2020-06-30)


### üêõ„ÄÄBug Fixes

* support more pinyin pattern ([8da3168](https://github.com/bluelovers/ws-regexp/commit/8da3168efdeac0c581eb537bb011b935e9effa30))


### ‚ú®„ÄÄFeatures

* tweak default options ([ece2695](https://github.com/bluelovers/ws-regexp/commit/ece2695fdcd3a81f7770dc339cf217f3359c0e04))
* add ``@regexp-cjk/regex-pinyin` ([56197e7](https://github.com/bluelovers/ws-regexp/commit/56197e73f20debadd08b70bf6dc4232139675dee))


### BREAKING CHANGES

* only set useTransliteration = true when pinyinMode, pinyinType is not set





## [1.0.42](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.41...@lazy-cjk/slugify@1.0.42) (2020-06-29)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.41](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.40...@lazy-cjk/slugify@1.0.41) (2020-06-28)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.40](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.39...@lazy-cjk/slugify@1.0.40) (2020-06-27)


### üîñ„ÄÄMiscellaneous

* . ([f70e3c2](https://github.com/bluelovers/ws-regexp/commit/f70e3c280ab4a271167353937be9749385e69d92))





## [1.0.39](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.38...@lazy-cjk/slugify@1.0.39) (2020-06-27)


### ‚ú®„ÄÄFeatures

* add `noSepBetweenZhChar` ([b2f431a](https://github.com/bluelovers/ws-regexp/commit/b2f431a2ace915ad50c45a268d6459245a7ce7fd))





## [1.0.38](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.37...@lazy-cjk/slugify@1.0.38) (2020-06-26)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.37](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.36...@lazy-cjk/slugify@1.0.37) (2020-06-20)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.36](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.35...@lazy-cjk/slugify@1.0.36) (2020-06-17)


### üîñ„ÄÄMiscellaneous

* . ([9fabaaf](https://github.com/bluelovers/ws-regexp/commit/9fabaafb0fe169844fc443c040966c28271a0f99))





## [1.0.35](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.34...@lazy-cjk/slugify@1.0.35) (2020-06-11)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.34](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.33...@lazy-cjk/slugify@1.0.34) (2020-06-11)


### Features

* @lazy-cjk/emoji-regex ([383372e](https://github.com/bluelovers/ws-regexp/commit/383372edc6be3918f66f890e40f898d273b5c2b7))





## [1.0.33](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.32...@lazy-cjk/slugify@1.0.33) (2020-06-09)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.32](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.31...@lazy-cjk/slugify@1.0.32) (2020-06-08)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.31](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.30...@lazy-cjk/slugify@1.0.31) (2020-06-08)


### Features

* allow config chinese pinyin options ([a0239d7](https://github.com/bluelovers/ws-regexp/commit/a0239d7b9a6cc222ed655da7c36d6557cd4dfb2c))





## [1.0.30](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.29...@lazy-cjk/slugify@1.0.30) (2020-06-07)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.29](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.28...@lazy-cjk/slugify@1.0.29) (2020-06-07)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.28](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.27...@lazy-cjk/slugify@1.0.28) (2020-06-06)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.27](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.26...@lazy-cjk/slugify@1.0.27) (2020-06-06)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.26](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.25...@lazy-cjk/slugify@1.0.26) (2020-06-06)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.25](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.24...@lazy-cjk/slugify@1.0.25) (2020-06-06)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.24](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.23...@lazy-cjk/slugify@1.0.24) (2020-06-05)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.23](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.22...@lazy-cjk/slugify@1.0.23) (2020-06-04)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.22](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.21...@lazy-cjk/slugify@1.0.22) (2020-06-04)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.21](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.20...@lazy-cjk/slugify@1.0.21) (2020-06-03)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.20](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.19...@lazy-cjk/slugify@1.0.20) (2020-06-03)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.19](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.18...@lazy-cjk/slugify@1.0.19) (2020-06-03)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.18](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.17...@lazy-cjk/slugify@1.0.18) (2020-06-02)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.17](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.16...@lazy-cjk/slugify@1.0.17) (2020-06-02)


### Features

* support slugify for korean japanese ([9b005fd](https://github.com/bluelovers/ws-regexp/commit/9b005fda87002ae09c86293d11cad30d22e69110))





## [1.0.16](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.15...@lazy-cjk/slugify@1.0.16) (2020-06-02)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.15](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.14...@lazy-cjk/slugify@1.0.15) (2020-06-01)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.14](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.13...@lazy-cjk/slugify@1.0.14) (2020-06-01)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.13](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.12...@lazy-cjk/slugify@1.0.13) (2020-06-01)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.12](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.11...@lazy-cjk/slugify@1.0.12) (2020-06-01)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.11](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.10...@lazy-cjk/slugify@1.0.11) (2020-05-31)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.10](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.9...@lazy-cjk/slugify@1.0.10) (2020-05-31)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.9](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.8...@lazy-cjk/slugify@1.0.9) (2020-05-31)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.8](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.7...@lazy-cjk/slugify@1.0.8) (2020-05-31)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.7](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.6...@lazy-cjk/slugify@1.0.7) (2020-05-31)

**Note:** Version bump only for package @lazy-cjk/slugify





## [1.0.6](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.5...@lazy-cjk/slugify@1.0.6) (2020-05-31)


### Features

* options.maxLength ([76fcfb0](https://github.com/bluelovers/ws-regexp/commit/76fcfb0a2d5da4b4a32bcb1f955fde7e22012040))





## [1.0.5](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.4...@lazy-cjk/slugify@1.0.5) (2020-05-31)


### Features

* options.upperCase ([388b6e7](https://github.com/bluelovers/ws-regexp/commit/388b6e7aab8e7e2770c2ebebfeacb6927e3ef4f6))





## [1.0.4](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.3...@lazy-cjk/slugify@1.0.4) (2020-05-31)


### Bug Fixes

* regexp ([0064dcd](https://github.com/bluelovers/ws-regexp/commit/0064dcdbdb30a51692f65bb5b9ce0168ae7f86b2))





## [1.0.3](https://github.com/bluelovers/ws-regexp/compare/@lazy-cjk/slugify@1.0.2...@lazy-cjk/slugify@1.0.3) (2020-05-31)


### Features

* @lazy-cjk/slugify ([f24aa90](https://github.com/bluelovers/ws-regexp/commit/f24aa907778bfdb61e1fdc1713949908a240ab21))





## 1.0.2 (2020-05-29)

**Note:** Version bump only for package @lazy-cjk/slugify





## 1.0.1 (2020-05-29)

**Note:** Version bump only for package @lazy-cjk/slugify
