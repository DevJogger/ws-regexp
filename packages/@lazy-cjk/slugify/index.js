"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.slugify = exports.transliterate = void 0;
const core_1 = require("./lib/core");
function transliterate(word, options) {
    options = (0, core_1.handleOptions)(options);
    options = {
        ...options,
    };
    options.separator = ' ';
    options.noSepBetweenZhChar = true;
    options.noStripOthers = true;
    return (0, core_1._core)(word, options);
}
exports.transliterate = transliterate;
function slugify(word, options) {
    options = (0, core_1.handleOptions)(options);
    return (0, core_1._core)(word, options);
}
exports.slugify = slugify;
exports.default = slugify;
//# sourceMappingURL=index.js.map