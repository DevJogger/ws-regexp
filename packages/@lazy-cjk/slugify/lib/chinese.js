"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.slugifyCjk = exports._replaceCjk = void 0;
const cns_11643_1 = require("@lazy-cjk/cns-11643");
const core_1 = require("./core");
const transliteration_1 = require("transliteration");
const cjk_conv_1 = require("regexp-helper/lib/cjk-conv");
const REGEXP_TEST = new RegExp(cjk_conv_1._re_cjk_conv('u').source, 'ug');
function _replaceCjk(text, options) {
    var _a;
    let append = (_a = options === null || options === void 0 ? void 0 : options.separator) !== null && _a !== void 0 ? _a : ' ';
    return text.replace(REGEXP_TEST, (s) => {
        let n = transliteration_1.slugify(s);
        if (n === '') {
            n = cns_11643_1.char2pinyin_01(s)[0];
        }
        if (n) {
            return n + append;
        }
        return s;
    });
    /*
    return replaceChar(text, (s, c) =>
    {
        let n = _slugify(c);

        if (n === '')
        {
            n = char2pinyin_01(c)[0]
        }

        return n + append
    })
     */
}
exports._replaceCjk = _replaceCjk;
function slugifyCjk(word, options) {
    return core_1._core(_replaceCjk(word), options);
}
exports.slugifyCjk = slugifyCjk;
//# sourceMappingURL=chinese.js.map