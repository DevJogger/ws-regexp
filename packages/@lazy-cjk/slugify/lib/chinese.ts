import { replaceChar, char2pinyin_01 } from '@lazy-cjk/cns-11643';
import { IOptionsSlugify } from './types';
import { _core } from './core';
import { slugify as _slugify } from 'transliteration';

export function _replaceChinese(text: string, options?: IOptionsSlugify)
{
	let append = options?.separator ?? ' ';

	return replaceChar(text, (s, c) =>
	{
		let n = _slugify(c);

		if (n === '')
		{
			n = char2pinyin_01(c)[0]
		}

		return n + append
	})
}

export function slugifyChinese(word: string, options?: IOptionsSlugify)
{
	return _core(_replaceChinese(word), options)
}
