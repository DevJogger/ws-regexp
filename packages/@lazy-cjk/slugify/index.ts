/**
 * Created by user on 2020/5/30.
 */
import { IOptionsSlugify } from './lib/types';
import { _core, handleOptions, _coreCase, _slice } from './lib/core';
import { _replaceEmoji } from './lib/emoji';
import { _replaceCjk } from './lib/chinese';
import _transliterate from '@sindresorhus/transliterate';

function _text(word: string, options: IOptionsSlugify)
{
	if (options.emoji)
	{
		word = _replaceEmoji(word, options);
	}

	if (options.cjk ?? true)
	{
		word = _replaceCjk(word, options);
	}

	if (options.transliterate ?? true)
	{
		word = _transliterate(word);
	}

	return word
}

export function transliterate(word: string, options?: IOptionsSlugify)
{
	options = handleOptions(options);
	options = {
		...options,
	}
	options.separator = ' ';

	word = _text(word, options);

	word = _coreCase(word, options);

	word = _slice(word, options);

	return word
		.replace(/\s+/g, ' ')
}

export function slugify(word: string, options?: IOptionsSlugify)
{
	options = handleOptions(options);

	word = _text(word, options);

	return _core(word, options)
}

export default slugify


