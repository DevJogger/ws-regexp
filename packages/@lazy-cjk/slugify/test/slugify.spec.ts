import slugify, { transliterate } from '../index';
import { IOptionsSlugify } from '../lib/types';

let word = `不支援 𠮷𠬠𡬶𫗭𣛙𢎐 ...這類字🍔\t🍕\t🍖\t🍗\t🍘\t🍙\t🍚\t🍛[Pixiv] Judo ユフォリア (5026941)(呉マサヒロ)] 高飛車な生徒会長をアプリで従順調教 ☆﹐﹑veevee˚ㆍ ㅔㅣ ㅑㅣㅗ므ㅛㅕㅇㅁ ㅇㅜㅇ ㅏㅏ Whiskers ˶˃ᆺ˂˶`;

_testFn(transliterate);
_testFn(slugify);

_testFn(transliterate, {
	emoji: true,
});
_testFn(slugify, {
	emoji: true,
});

_testFn(transliterate, {
	noStripOthers: true,
});
_testFn(slugify, {
	noStripOthers: true,
});

function _testFn(fn: typeof transliterate | typeof slugify, opts: IOptionsSlugify = {})
{
	test(`${fn.name}: ${JSON.stringify(opts)}`, () =>
	{
		let actual = fn(word, opts);

		console.dir({
			word,
			opts,
		});
		console.log(actual);

		expect(actual).not.toStrictEqual(word);

		expect(actual).toMatchSnapshot();
	})
}
