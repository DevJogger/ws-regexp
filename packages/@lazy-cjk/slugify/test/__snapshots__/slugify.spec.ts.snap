// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`slugify: {"emoji":true} 1`] = `"bu zhi yuan  jí mǒu xín wèi shuò shàng   zhe lei zi hamburger pizza meat on bone poultry leg rice cracker rice ball cooked rice curry rice Pixiv  Judo yuforia  5026941 wu masahiro  gao fei che nasheng tu hui chang oapuridecong shun diao jiao   veevee     meu m     Whiskers"`;

exports[`slugify: {"noStripOthers":true} 1`] = `"bu-zhi-yuan- jí-mǒu-xín-wèi-shuò-shàng- ...zhe-lei-zi-🍔 🍕 🍖 🍗 🍘 🍙 🍚 🍛[Pixiv] Judo yuforia (5026941)(wu-masahiro)] gao-fei-che-nasheng-tu-hui-chang-oapuridecong-shun-diao-jiao- ☆﹐﹑veevee˚ㆍ ㅔㅣ ㅑㅣㅗmeuㅛㅕm ㅜ ㅏㅏ Whiskers ˶˃ᆺ˂˶"`;

exports[`slugify: {} 1`] = `"bu zhi yuan  jí mǒu xín wèi shuò shàng   zhe lei zi Pixiv  Judo yuforia  5026941 wu masahiro  gao fei che nasheng tu hui chang oapuridecong shun diao jiao   veevee     meu m     Whiskers"`;

exports[`transliterate: {"emoji":true} 1`] = `"buzhiyuan jímǒuxínwèishuòshàng ...zheleizihamburger  pizza  meat on bone  poultry leg  rice cracker  rice ball  cooked rice  curry rice [Pixiv] Judo yuforia (5026941)(wumasahiro)] gaofeichenashengtuhuichangoapuridecongshundiaojiao ☆﹐﹑veevee˚ㆍ ㅔㅣ ㅑㅣㅗmeuㅛㅕm ㅜ ㅏㅏ Whiskers ˶˃ᆺ˂˶"`;

exports[`transliterate: {"noStripOthers":true} 1`] = `"buzhiyuan jímǒuxínwèishuòshàng ...zheleizi🍔 🍕 🍖 🍗 🍘 🍙 🍚 🍛[Pixiv] Judo yuforia (5026941)(wumasahiro)] gaofeichenashengtuhuichangoapuridecongshundiaojiao ☆﹐﹑veevee˚ㆍ ㅔㅣ ㅑㅣㅗmeuㅛㅕm ㅜ ㅏㅏ Whiskers ˶˃ᆺ˂˶"`;

exports[`transliterate: {} 1`] = `"buzhiyuan jímǒuxínwèishuòshàng ...zheleizi🍔 🍕 🍖 🍗 🍘 🍙 🍚 🍛[Pixiv] Judo yuforia (5026941)(wumasahiro)] gaofeichenashengtuhuichangoapuridecongshundiaojiao ☆﹐﹑veevee˚ㆍ ㅔㅣ ㅑㅣㅗmeuㅛㅕm ㅜ ㅏㅏ Whiskers ˶˃ᆺ˂˶"`;
